Userland/Libraries/LibJS/Runtime/RegExpConstructor.h

```diff
@@ -6,10 +6,70 @@
 
 #pragma once
 
+#include <AK/HashMap.h>
+#include <AK/Utf16View.h>
 #include <LibJS/Runtime/NativeFunction.h>
 
 namespace JS {
 
+// https://github.com/tc39/proposal-regexp-legacy-features#regexp
+// The %RegExp% instrinsic object, which is the builtin RegExp constructor, has the following additional internal slots:
+// [[RegExpInput]]
+// [[RegExpLastMatch]]
+// [[RegExpLastParen]]
+// [[RegExpLeftContext]]
+// [[RegExpRightContext]]
+// [[RegExpParen1]] ... [[RegExpParen9]]
+class RegExpLegacyStaticProperties {
+public:
+    Optional<String> input() const { return m_input; }
+    Optional<String> last_match() const { return m_last_match; }
+    Optional<String> last_paren() const { return m_last_paren; }
+    Optional<String> left_context() const { return m_left_context; }
+    Optional<String> right_context() const { return m_right_context; }
+    Optional<String> $1() const { return m_$1; }
+    Optional<String> $2() const { return m_$2; }
+    Optional<String> $3() const { return m_$3; }
+    Optional<String> $4() const { return m_$4; }
+    Optional<String> $5() const { return m_$5; }
+    Optional<String> $6() const { return m_$6; }
+    Optional<String> $7() const { return m_$7; }
+    Optional<String> $8() const { return m_$8; }
+    Optional<String> $9() const { return m_$9; }
+
+    void set_input(String const& input) { m_input = move(input); }
+    void set_last_match(String const& last_match) { m_last_match = move(last_match); }
+    void set_last_paren(String const& last_paren) { m_last_paren = move(last_paren); }
+    void set_left_context(String const& left_context) { m_left_context = move(left_context); }
+    void set_right_context(String const& right_context) { m_right_context = move(right_context); }
+    void set_$1(String const& value) { m_$1 = move(value); }
+    void set_$2(String const& value) { m_$2 = move(value); }
+    void set_$3(String const& value) { m_$3 = move(value); }
+    void set_$4(String const& value) { m_$4 = move(value); }
+    void set_$5(String const& value) { m_$5 = move(value); }
+    void set_$6(String const& value) { m_$6 = move(value); }
+    void set_$7(String const& value) { m_$7 = move(value); }
+    void set_$8(String const& value) { m_$8 = move(value); }
+    void set_$9(String const& value) { m_$9 = move(value); }
```

Comment:
These all should take by value (`set_foo(String value)`) for the `move()` to actually take effect